src/config.rs

```diff
@@ -48,6 +46,66 @@ base64_serde_type!(Base64Standard, base64::STANDARD);
 // of every `LOG_SAMPLING_RATE` occurrences to avoid spamming the logs.
 pub(crate) const LOG_SAMPLING_RATE: u64 = 1000;
 
+/// Config is the configuration of a proxy
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+#[non_exhaustive]
+pub struct Config {
+    pub version: Version,
+
+    #[serde(default)]
+    pub proxy: Proxy,
+
+    #[serde(default)]
+    pub admin: Admin,
+
+    #[serde(flatten)]
+    pub source: Source,
+}
+
+impl Config {
+    /// Attempts to locate and parse a `Config` located at either `path`, the
+    /// `$QUILKIN_CONFIG` environment variable if set, the current directory,
+    /// or the `/etc/quilkin` directory (on unix platforms only). Returns an
+    /// error if the found configuration is invalid, or if no configuration
+    /// could be found at any location.
+    pub fn find(
+        log: &slog::Logger,
```

Comment:
Having to accept `Logger` here is unfortunate, but once #317 is implemented, this argument should be able to be removed.